src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx

```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
+    [marginAccountBalance.fPositionBC, perpParameters, trade],
+  );
+
+  const tradingFee = useMemo(
+    () => (trade ? getTradingFee(Number(trade!.amount), perpParameters) : 0),
+    [perpParameters, trade],
+  );
+
+  const isBuy = useMemo(() => trade?.position === TradingPosition.LONG, [
+    trade?.position,
+  ]);
+
+  if (!trade) {
+    return null;
+  }
+
   return (
     <Dialog
       isOpen={modal === PerpetualPageModals.TRADE_REVIEW}
       onClose={onClose}
     >
       <h1>{t(translations.perpetualPage.reviewTrade.title)}</h1>
-      {/* TODO: implement Review Trade Dialog */}
-      <pre>{JSON.stringify(trade, null, 2)}</pre>
-      <button onClick={onSubmit}>Open trade</button>
+      <div className={styles.contentWrapper}>
+        <div className={styles.importantInfo}>
+          <div
+            className={classNames(
+              'tw-text-xl tw-font-medium tw-tracking-normal tw-leading-none',
+              isBuy ? styles.orderActionBuy : styles.orderActionSell,
+            )}
+          >
+            {trade?.leverage}x{' '}
+            {t(translations.perpetualPage.reviewTrade.market)}{' '}
+            {getTradePosition(trade?.position)}
+          </div>
+          <div className={styles.orderActionAdditionalInfo}>
+            {toNumberFormat(fromWei(trade?.amount), 3)} BTC @{' '}
+            {isBuy ? '≥' : '≤'} {toNumberFormat(trade.entryPrice, 2)} USD
+          </div>
+        </div>
+
+        <div className={classNames(styles.importantInfo, 'tw-mt-4')}>
+          <div className={styles.tradingFeeWrapper}>
+            <span className="tw-text-gray-10 tw-font-light">
+              {t(translations.perpetualPage.tradeForm.labels.tradingFee)}
+            </span>
+            <span className="tradingFeeValue">
+              <AssetValue
+                minDecimals={0}
+                maxDecimals={6}
+                mode={AssetValueMode.auto}
+                value={String(tradingFee)}
+                assetString="BTC"
+              />
+            </span>
+          </div>
+        </div>
+
+        <div className={styles.positionDetailsTitle}>New Position Details</div>
+
+        <div className={styles.positionInfo}>
+          <div className={styles.positionInfoRow}>
+            <span className="tw-text-gray-10 tw-font-light">
+              {t(translations.perpetualPage.reviewTrade.labels.positionSize)}
+            </span>
+            <span className={styles.positionSize}>
+              +{toNumberFormat(fromWei(trade?.amount), 3)} BTC
+            </span>
+          </div>
+
+          <div className={classNames(styles.positionInfoRow, 'tw-mt-2')}>
+            <span className="tw-text-gray-10 tw-font-light">
+              {t(translations.perpetualPage.reviewTrade.labels.margin)}
+            </span>
+            <span>
+              <AssetValue
+                minDecimals={4}
+                maxDecimals={4}
+                mode={AssetValueMode.auto}
+                value={String(margin)}
```

Comment:
margin is not a wei amount so pass it as number instead.
```suggestion
                value={margin}
```

Reply:
It's in wei but it's not a string. It's used the same way in `TradeForm`.